### PR TITLE
Update to grade 7 and spark 3.2

### PR DESCRIPTION
**Pull request by adahn6**
### Description:

What the title says! may require updates to publishing logic if I didn't get it right the first time 😬 

### Potential Impact:

Hopefully pretty limited, this is a dependency update and build update more than anything else

### Test Results:

------

In doubt: [Contributing Guidelines](https://github.com/osmlab/atlas/blob/dev/CONTRIBUTING.md)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -60,8 +47,8 @@ configurations
             force packages.snappy
         }
     }
-    shaded
-    {
+    shaded {
+        extendsFrom(implementation)
```
Comment (matthieun) [inline build.gradle:51]: Nice

File: build.gradle [hunk 2]
```diff
@@ -162,3 +148,9 @@ idea {
         languageLevel = '11'
     }
 }
+
+tasks.each {
+  task -> if (task.hasProperty("duplicatesStrategy")) {
```
Comment (matthieun) [inline build.gradle:153]: What does that do?

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionSha256Sum=8b356fd8702d5ffa2e066ed0be45a023a779bba4dd1a68fd11bc2a6bdc981e8f
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.9.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0-bin.zip
```
Comment (matthieun) [inline gradle/wrapper/gradle-wrapper.properties:3]: Why not use the latest, `7.5.1` ?

**Review by matthieun — CHANGES_REQUESTED** [review]
Awesome! Just two questions.. Thanks!

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (adahn6) [reply gradle/wrapper/gradle-wrapper.properties]: I thought I did! switched to it now

File: build.gradle (hunk 2 above)
Comment (adahn6) [reply build.gradle:153]: this is just an annoyance, sometimes there are resources with the same name especially in tests and gradle complains. we have this same fix in atlas too 🤷‍♀️

**Review by matthieun — DISMISSED** [review]

**Comment by matthieun** [comment]
Seems like Checkstyle is unhappy:
```
Error: eckstyle] [ERROR] /home/runner/work/atlas-generator/atlas-generator/src/main/java/org/openstreetmap/atlas/generator/tools/spark/rdd/ShardedAtlasRDDLoader.java:94:8: Unable to get class information for @throws tag 'CoreException'. [JavadocMethod]
```

**Review by matthieun — APPROVED** [review]

**Pull request merged**